File: src/main/scala/com/abaddon83/burraco/match/games/domainModels/Scale.scala
package com.abaddon83.burraco.`match`.games.domainModels

import com.abaddon83.burraco.shares.decks.Suits.Suit
import com.abaddon83.burraco.shares.decks.{Card, Ranks}


trait Scale {
  protected val cards: List[Card]
  protected val suit: Suit


  def addCards(cardsToAdd: List[Card]): Scale

  protected def validateNewCards(cards: List[Card]): Scale

}

File: src/main/kotlin/com/abaddon83/burracoGame/writeModel/models/burracos/Scale.kt
package com.abaddon83.burracoGame.writeModel.models.burracos

import com.abaddon83.burracoGame.writeModel.models.decks.Card
import com.abaddon83.burracoGame.writeModel.models.decks.Ranks
import com.abaddon83.burracoGame.writeModel.models.decks.Suits

abstract class Scale(className:String): Burraco(className) {

    fun numCards():Int = cards.size
    fun showSuit(): Suits.Suit = suit

    abstract fun addCards(cardsToAdd: List<Card>): Scale
    protected abstract val suit: Suits.Suit
    protected abstract fun validateNewCards(cardsToValidate: List<Card>): List<Card>

}